server/metadata/tenant.go

```diff
@@ -201,12 +201,28 @@ func (m *TenantManager) GetNamespace(ctx context.Context, namespaceName string,
 	m.RUnlock()
 
 	m.Lock()
-	defer m.Unlock()
 	if tenant, found := m.tenants[namespaceName]; found {
 		return tenant.namespace, nil
 	}
-	// this will never create new namespace
-	// this is for reading namespaces from storage into cache
+	m.Unlock()
+	namespaces, err := m.ListNamespaces(ctx, txMgr)
+	if err != nil {
+		return nil, err
+	}
+	if namespaces == nil {
+		return nil, api.Errorf(api.Code_NOT_FOUND, "Namespace not found")
+	}
+	for i := range namespaces {
+		if namespaces[i].Name() == namespaceName {
+			return namespaces[i], nil
+		}
+	}
+	return nil, api.Errorf(api.Code_NOT_FOUND, "Namespace not found")
+}
+
+func (m *TenantManager) ListNamespaces(ctx context.Context, txMgr *transaction.Manager) ([]Namespace, error) {
```

Comment:
as you're controlling the transaction for the create namespace in the service layer, should we do the same for ListNamespace? The GetNamespace will create the transaction and pass it to ListNamespace.

Reply:
made it consistent.